### PR TITLE
NHShealthierfamilies ingredient grouping support

### LINKED ISSUE #884: Bug: NHS Healthier Families: optional additional ingredients are missing
**Pre-filing checks**

- [x] I have searched for open issues that report the same problem
- [x] I have checked that the bug affects the latest version of the library

**The URL of the recipe(s) that are not being scraped correctly**

- https://www.nhs.uk/healthier-families/recipes/homemade-fish-and-chips/

**The results you expect to see**

Mushy peas and ground black pepper _could_ appear in the ingredient list - although... as the recipe mentions, they're optional/swappable.  The interface we have doesn't really provide a distinction between required and optional ingredients at the moment.

**The results (including any Python error messages) that you are seeing**

```pyi
>>> from recipe_scrapers import scrape_me
>>> s = scrape_me('https://www.nhs.uk/healthier-families/recipes/homemade-fish-and-chips/')
>>> s.ingredients()
['4 potatoes, scrubbed, each cut into 8 wedges', '1 tablespoon vegetable oil', '75g dried white or wholemeal breadcrumbs', '1 egg, beaten with 2 tbsp cold water', '4 fillets skinless white fish, like haddock, cod or pollock']
```

### PR DESCRIPTION
**Pull request by jknndy**
Resolves #884 , added functionality to find all li until it encounters an ol.

Adds ingredient grouping support as well

### CODE REVIEW

**Comment by jknndy** [comment]
I also noticed this site isn't listed in the support sites list, I can add it with this PR unless its excluded on purpose?

**Comment by jayaddison** [comment]
Thanks!  I'm going to try to test this out on one or two more recipes.

And nope, no reason it was omitted from the website list (I'm pretty sure I wrote the original scraper).  I tend to use an extremely manual, low-tech development approach (not using the generator code usually) and that'll be why I forgot to add that entry.

File: tests/test_nhshealthierfamilies_2.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                        "4 fillets skinless white fish, like haddock, cod or pollock",
+                    ],
+                    purpose=None,
+                ),
+                IngredientGroup(
+                    ingredients=[
+                        "300g mushy peas",
+                        "1 pinch ground black pepper (optional)",
+                    ],
+                    purpose="Swappable or optional",
```
Comment (jayaddison) [inline tests/test_nhshealthierfamilies_2.py:62]: Kind of a nitpick/philosophical thought: in most recipes the groupings are for subcomponents of the meal used during preparation/cooking.  In this case, this grouping is for some extra items that are each individually nice-to-haves.

I think it's fine - in fact I think ingredient groups are a good solution here.  It's mostly an observation that I wanted to mention.

Comment (jknndy) [reply tests/test_nhshealthierfamilies_2.py:62]: I made sure to check against another recipe that was more inline with the actual purpose and everything works. I guess this is just a new use-case for it!

https://www.nhs.uk/healthier-families/recipes/bengali-chicken-curry/

Comment (jayaddison) [reply tests/test_nhshealthierfamilies_2.py:62]: Neat :)  thanks!

**Review by jayaddison — APPROVED** [review]

**Comment by jayaddison** [comment]
Looks good to me - merging :+1:

**Pull request merged**